services/checks/checks.go

```diff
@@ -104,19 +109,30 @@ type Service struct {
 }
 
 // New returns Service with given PMM version.
-func New(agentsRegistry agentsRegistry, alertsRegistry alertRegistry, db *reform.DB) *Service {
+func New(agentsRegistry agentsRegistry, alertmanagerService alertmanagerService, db *reform.DB) *Service {
 	l := logrus.WithField("component", "checks")
 
-	s := &Service{
-		agentsRegistry: agentsRegistry,
-		alertsRegistry: alertsRegistry,
-		db:             db,
+	var resendInterval time.Duration
+	if d, err := time.ParseDuration(os.Getenv(envResendInterval)); err == nil && d > 0 {
+		l.Warnf("Interval changed to %s.", d)
+		resendInterval = d
+	} else {
+		resendInterval = defaultResendInterval
+	}
 
-		l:          l,
-		host:       defaultHost,
-		publicKeys: defaultPublicKeys,
-		interval:   defaultInterval,
-		startDelay: defaultStartDelay,
+	s := &Service{
+		agentsRegistry:      agentsRegistry,
+		alertmanagerService: alertmanagerService,
+		db:                  db,
+		alertsRegistry:      newRegistry(),
+
+		l:              l,
+		host:           defaultHost,
+		publicKeys:     defaultPublicKeys,
+		interval:       defaultInterval,
+		startDelay:     defaultStartDelay,
+		resendInterval: resendInterval,
+		alertTTL:       resolveTimeoutFactor * resendInterval,
```

Comment:
I think better to pass that value to registry constructor on line 127.

Reply:
done